### PR TITLE
Delete Kibana client and Fleet API code from E2E testing framework

### PR DESCRIPTION
**Pull request by ycombinator**
## What does this PR do?

This PR deletes the code for the Kibana client and the Fleet APIs from the E2E testing framework.

## Why is it important?

The code has been ported over to the existing Kibana client we have in `elastic-agent-libs`; see https://github.com/elastic/elastic-agent-libs/pull/118.  This prevents us from ending up with two Kibana clients in the Agent codebase.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [x] I have added an integration test or an E2E test

## Related issues

- https://github.com/elastic/elastic-agent-libs/pull/118

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -165,3 +165,7 @@ replace (
 
 // Exclude this version because the version has an invalid checksum.
 exclude github.com/docker/distribution v2.8.0+incompatible
+
+replace (
+	github.com/elastic/elastic-agent-libs => github.com/ycombinator/elastic-agent-libs v0.0.0-20230504062120-4ef5f2463ba0
+)
```
Comment (ycombinator) [inline go.mod:171]: These lines should be removed and the `require` directive for `elastic-agent-libs` should be update once https://github.com/elastic/elastic-agent-libs/pull/118 is merged and built.

**Comment by ycombinator** [comment]
I'm not sure why the Check stage is failing in CI.  I've run `make notice` and committed the changes.

File: go.mod [hunk 2]
```diff
@@ -12,12 +11,14 @@ require (
 	github.com/coreos/go-systemd/v22 v22.3.3-0.20220203105225-a9a7ef127534
 	github.com/docker/go-units v0.5.0
 	github.com/dolmen-go/contextio v0.0.0-20200217195037-68fc5150bcd5
+	github.com/elastic/beats/v7 v7.17.10
```
Comment (michalpristas) [inline go.mod:14]: is that right?

Comment (ycombinator) [reply go.mod:14]: No, this doesn't look right to me, thanks for catching.  No idea why it got added - same goes for some of the other changes in `go.mod`.  Let me try to revert these manually and see what happens.

**Review by AndersonQ — APPROVED** [review]

**Review by blakerouse — APPROVED** [review]
Looks good. Be sure to fix the go.mod replace before merging ;-)

I also think that we should move `e2e/tools` into `pkg/testing/tools`. Now that the kibana client is moved out I think that those tools are generally useful for test writers, so we should just make it a general module in `pkg/testing`.

**Review by michalpristas — APPROVED** [review]

**Comment by ycombinator** [comment]
> Looks good. Be sure to fix the go.mod replace before merging ;-)

Yes, thanks for the reminder.

> I also think that we should move `e2e/tools` into `pkg/testing/tools`. Now that the kibana client is moved out I think that those tools are generally useful for test writers, so we should just make it a general module in `pkg/testing`.

++ I'll do this in a follow up PR.

File: go.mod (hunk 1 above)
Comment (ycombinator) [reply go.mod]: Removed in 219eadd81bb335211f015cbf7858efb946b941a3.

**Pull request merged**